### PR TITLE
add mask to matching function

### PR DESCRIPTION
**Pull request by ChrisRBXiong**
add mask to matching function, and modify match_pyramid to adapt to it

### CODE REVIEW

File: matchzoo/models/match_pyramid.py [hunk 1]
```diff
@@ -111,9 +113,16 @@ def forward(self, inputs):
...
+        # Left and right input mask matrix
+        # shape = [B, L]
+        # shape = [B, R]
+        left_mask = (input_left == self._params['mask_value'])
+        right_mask = (input_right == self._params['mask_value'])
+
         # Compute matching signal
         # shape = [B, 1, L, R]
-        embed_cross = self.matching(embed_left, embed_right).unsqueeze(dim=1)
+        embed_cross = self.matching(embed_left, embed_right, left_mask, right_mask
```
Comment (Chriskuei) [inline matchzoo/models/match_pyramid.py]:
```suggestion
        embed_cross = self.matching(
            embed_left, embed_right, left_mask, right_mask
         ).unsqueeze(dim=1)
```

File: matchzoo/modules/matching.py [hunk 2]
```diff
@@ -40,30 +40,44 @@ def _validate_matching_type(cls, matching_type: str = 'dot'):
...
                              f"{valid_matching_type} expected.")
 
-    def forward(self, x, y):
+    def forward(self, x, y, x_mask=None, y_mask=None):
         """Perform attention on the input."""
         length_left = x.shape[1]
         length_right = y.shape[1]
         if self._matching_type == 'dot':
             if self._normalize:
                 x = F.normalize(x, p=2, dim=-1)
```
Comment (caiyinqiong) [inline matchzoo/modules/matching.py]: `mask` should be processed before `normalize`.
